src/UnitTests/Bug/ShouldMapCtor.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System.Linq;
+using Shouldly;
+using Xunit;
+
+namespace AutoMapper.UnitTests.Bug
+{
+    public abstract class ShouldMapCtor
```

Comment:
Or you can simply rename the namespace from Bug to ShouldMapCtor and then this class is not useful anymore. The file should be one level up. The tests should work against the API. Either assert that some constructor is used when calling Map or assert that configuration validation fails.

Reply:
I'll move it.

I wasn't comfortable with these tests, but couldn't get past the way I'd tried to work around outside of AM. Asserting the configuration validation fails, should be fine. I'll do that.